qualification/report/generate_pdf.py

```diff
@@ -835,13 +835,42 @@ def document_from_json(input_data: str | list) -> Document:
     return doc
 
 
+def test_image_commit(input_data: str | list) -> str:
+    """Extract the git commit ID of the test image from the test results.
+
+    Parameters
+    ----------
+    input_data: str | list
+        Either the list of parsed JSON entries, or a path to the report file
+        generated by :samp:`pytest --report-log={filename}`.
+
+    Returns
+    -------
+    commit_id: str
+        The git commit ID of the test image.
+    """
+    if not isinstance(input_data, list):
+        result_list = []
+        with open(input_data) as fp:
+            for line in fp:
+                result_list.append(json.loads(line))
+    else:
+        result_list = input_data
+    test_configuration = parse(result_list)[0]
+    commit_id = test_configuration.correlators[0].tasks["f.antenna_channelised_voltage.0"].git_version
```

Comment:
This is only considering one task of one correlator. If an update is pushed halfway through the qualification test, there could be an inconsistent mix of versions. I think it could also crash if the correlator doesn't start up successfully and so there is never any information about that task. Something like the following should work I think:
```python
git_versions = set()
for correlator in test_configuration.correlators:
    for task_name, task in correlator.tasks.items():
        if task_name != "product_controller"
            git_versions.add(task.git_version)
return collapse_versions(git_versions)
```